src/common/com/intellij/plugins/haxe/ide/info/HaxeParameterDescription.java

```diff
@@ -38,12 +39,16 @@ public HaxeParameterDescription(String name, String text, String varInitText) {
   }
 
 
-  public static HaxeParameterDescription[] getParameters(HaxeNamedComponent element, HaxeGenericSpecialization specialization) {
+  public static HaxeParameterDescription[] getParameters(HaxeNamedComponent element, HaxeGenericSpecialization specialization, boolean isExtension) {
     final HaxeParameterList parameterList = PsiTreeUtil.getChildOfType(element, HaxeParameterList.class);
     if (parameterList == null) {
       return new HaxeParameterDescription[0];
     }
-    final List<HaxeParameter> list = parameterList.getParameterList();
+    List<HaxeParameter> list = parameterList.getParameterList();
+    if (isExtension) {
+      list = new ArrayList<HaxeParameter>(list);
+      list.remove(0);
```

Comment:
The list that comes back from parameterList.getParameterList is already an ephemeral list.  It does not need to be copied to be manipulated.